roachtest: re-enable costfuzz test after fixing failures

This issue captures the first half-dozen failures of the new costfuzz randomized test. We need to sort these out before re-enabling the test. These are the individual failures that reproduce:

- https://github.com/cockroachdb/cockroach/issues/83023
- https://github.com/cockroachdb/cockroach/issues/83024
- https://github.com/cockroachdb/cockroach/issues/83025

Epic: CRDB-14914

---

roachtest.costfuzz [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/5250343?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/5250343?buildTab=artifacts#/costfuzz) on master @ [48e48db89eb3cd6bc38f3631364c516181811280](https://github.com/cockroachdb/cockroach/commits/48e48db89eb3cd6bc38f3631364c516181811280):


```
		  |   		... // 165 identical bytes
		  |   	}, ""),
		  |   	strings.Join({
		  |   		"1971-05-27 00:00:00 +0000 +0000,NULL,-7496,NL0A,-17571395363545.",
		  |   		"11588,390532174,{<\x1c,()+\x02,H,79 years 5 mons 236 days 11:24:41.412",
		  |   		"351,-0.4",
		  | - 		"812671945262518",
		  | + 		"3667785161269623",
		  |   		",-6 years -11 mons -979 days -05:53:34.05322,1151812650090379135",
		  |   		"9209.9247743890145185,104,3144597984,!,NULL,NULL,-4028770426.052",
		  |   		... // 177 identical bytes
		  |   	}, ""),
		  |   	... // 41 identical, 1 removed, 1 inserted, and 54 modified elements
		  |   }
		  | sql: SELECT
		  | 	tab_75349.col2_5 AS col_160035,
		  | 	NULL AS col_160036,
		  | 	(-7496):::INT8 AS col_160037,
		  | 	'NL0A':::STRING AS col_160038,
		  | 	(-17571395363545.11588):::DECIMAL AS col_160039,
		  | 	390532174:::OID AS col_160040,
		  | 	e'{<\x1c':::STRING AS col_160041,
		  | 	((NULL,) || tab_75350.col1_9::STRING)::STRING AS col_160042,
		  | 	'H':::STRING AS col_160043,
		  | 	tab_75350.col1_5 AS col_160044,
		  | 	lead(tab_75349.col2_1::FLOAT8) OVER (PARTITION BY tab_75350.col1_7, tab_75350.col1_1 ORDER BY tab_75350.col1_4 DESC, tab_75349.col2_4 ASC, tab_75350.col1_7 ASC, tab_75350.col1_12 GROUPS UNBOUNDED PRECEDING)::FLOAT8
		  | 		AS col_160045,
		  | 	tab_75350.col1_7 AS col_160046,
		  | 	tab_75350.col1_10 AS col_160047,
		  | 	tab_75350.tableoid AS col_160048,
		  | 	3144597984:::OID AS col_160049,
		  | 	'!':::STRING AS col_160050,
		  | 	NULL AS col_160051,
		  | 	NULL AS col_160052,
		  | 	(-4028770426.052056588):::DECIMAL AS col_160053,
		  | 	'1986-02-02 11:04:55.000131+00:00':::TIMESTAMPTZ AS col_160054,
		  | 	tab_75349.col2_3 AS col_160055,
		  | 	tab_75350.tableoid AS col_160056,
		  | 	B'11001100000111110110001111011110' AS col_160057,
		  | 	tab_75349.col2_1 AS col_160058,
		  | 	(-1.234E+401):::DECIMAL AS col_160059,
		  | 	(-6.611227290484173853E+31):::DECIMAL AS col_160060
		  | FROM
		  | 	defaultdb.public.table2@table2_col2_7_col2_0_col2_3_col2_5_col2_1_expr_col2_2_col2_6_idx AS tab_75349,
		  | 	defaultdb.public.table1@[0] AS tab_75350
		  | WHERE
		  | 	true
		  | ORDER BY
		  | 	tab_75349.tableoid, tab_75349.tableoid
		Error types: (1) *withstack.withStack (2) *errutil.leafError
```
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)

</p>
</details>
/cc @cockroachdb/sql-queries
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*costfuzz.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-16040

roachtest.costfuzz [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/5305827?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/5305827?buildTab=artifacts#/costfuzz) on master @ [e6815947a050e32f21e983aa30dc74ab2a247af3](https://github.com/cockroachdb/cockroach/commits/e6815947a050e32f21e983aa30dc74ab2a247af3):


```
The test failed on branch=master, cloud=gce:
test artifacts and logs in: /artifacts/costfuzz/run_1
	costfuzz.go:181,costfuzz.go:74,test_runner.go:884: error while printing perturbed statement results: pq: query execution canceled due to statement timeout
		(1) attached stack trace
		  -- stack trace:
		  | github.com/cockroachdb/cockroach/pkg/cmd/roachtest/tests.runCostFuzzQuery
		  | 	github.com/cockroachdb/cockroach/pkg/cmd/roachtest/tests/costfuzz.go:251
		  | github.com/cockroachdb/cockroach/pkg/cmd/roachtest/tests.runOneRoundCostFuzz
		  | 	github.com/cockroachdb/cockroach/pkg/cmd/roachtest/tests/costfuzz.go:180
		  | github.com/cockroachdb/cockroach/pkg/cmd/roachtest/tests.runCostFuzz
		  | 	github.com/cockroachdb/cockroach/pkg/cmd/roachtest/tests/costfuzz.go:74
		  | main.(*testRunner).runTest.func2
		  | 	main/pkg/cmd/roachtest/test_runner.go:884
		  | runtime.goexit
		  | 	GOROOT/src/runtime/asm_amd64.s:1581
		Wraps: (2) error while printing perturbed statement results
		Wraps: (3) pq: query execution canceled due to statement timeout
		Error types: (1) *withstack.withStack (2) *errutil.withPrefix (3) *pq.Error
```
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)

</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*costfuzz.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

--- a/pkg/cmd/roachtest/tests/costfuzz.go
+++ b/pkg/cmd/roachtest/tests/costfuzz.go
@@ -33,7 +33,6 @@ func registerCostFuzz(r registry.Registry) {
 		Tags:            nil,
 		Cluster:         r.MakeClusterSpec(1),
 		Run:             runCostFuzz,
-		Skip:            "flaky test: https://github.com/cockroachdb/cockroach/issues/81717",
 	})
 }
 

